src/umap/umap.cpp

```diff
@@ -301,23 +310,37 @@ void* umap_ex(void* base_addr, uint64_t region_size, int prot, int flags,
     return UMAP_FAILED;
   }
 
-  void* region = mmap(base_addr, region_size, prot, flags | (MAP_ANONYMOUS | MAP_NORESERVE), -1, 0);
+  //
+  // When dealing with umap-page-sizes that could be multiples of the actual
+  // system-page-size, it is possible for mmap() to provide a region that is on
+  // a system-page-boundary, but not necessarily on a umap-page-size boundary.
+  //
+  // We always allocate an additional umap-page-size set of bytes so that we can
+  // make certain that the umap-region begins on a umap-page-size boundary.
+  //
+  uint64_t mmap_size = region_size + umapPageSize;
+
+  void* mmap_region = mmap(base_addr, mmap_size, 
+                        prot, flags | (MAP_ANONYMOUS | MAP_NORESERVE), -1, 0);
```

Comment:
Is the MAP_ALIGN option of mmap available for this call? Then mmap would return an address that is aligned to the requested umap page size. I couldn't tell from the documentation if this option can be used for MAP_ANONYMOUS.

Reply:
I am not familiar with this option and don't see it on the manual page on mmap on rzansel.

Reply:
OK, it turns out that option is in the freebsd version of mmap and not the linux version. too bad -- it would really simplify the code.